packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java

```diff
@@ -146,42 +152,67 @@ private static Bundle convertArguments(Map<String, ?> arguments) {
         bundle.putLongArray(key, (long[]) value);
       } else if (value instanceof double[]) {
         bundle.putDoubleArray(key, (double[]) value);
-      } else if (isTypedArrayList(value, Integer.class)) {
-        bundle.putIntegerArrayList(key, (ArrayList<Integer>) value);
-      } else if (isTypedArrayList(value, String.class)) {
-        bundle.putStringArrayList(key, (ArrayList<String>) value);
-      } else if (isStringKeyedMap(value)) {
-        bundle.putBundle(key, convertArguments((Map<String, ?>) value));
+      } else if (integers(value) != null) {
```

Comment:
nit: I think I prefer the old name of these - I'm a bit surprised to see a function named `integers` or `strings` or `map`.

I also wonder if we can just somehow avoid traversing the array twice and creating the object twice.